### PR TITLE
Implement new data source: ´herokux_addons´

### LINKED ISSUE #72: New data source: herokux_addons
Hello there,

I'd like to be able to get the list of **Postgres/Redis** add-ons installed in my Heroku app. This would make it easier to 
set the maintenance window to Postgres/Redis add-ons because I wouldn't need to know the ID of every single add-on. 

Context: I've got dozens of Heroku apps and some of them have more than 1 Postgres/Redis add-ons installed, so it's kinda difficult to get the ID of each add-on. A data source could make things much easier, for example:

```
data "herokux_addons" "redis_addons" {
  app_id = "MY_APP_ID"
  addon_service_name = "heroku-redis"
}

resource "herokux_redis_maintenance_window" "redis_addons" {
  # Set the maintenance window for each add-on
  count = length(data.herokux_addons.redis_addons.addon_ids)

  redis_id = data.herokux_addons.redis_addons.addon_ids[count.index]
  window = "Sundays 07:00"
}
```

### PR DESCRIPTION
**Pull request by marcelomilera**
Implementing new data source `herokux_addons` that gets the list of add-ons installed on a particular Heroku app.

Resolves #72

### CODE REVIEW

File: herokux/provider.go
Comment (davidji99) [inline herokux/provider.go:253]: Please move this up above `herokux_postgres_mtls_certificate` so the list is in alphabetical order.

**Comment by davidji99** [comment]
THanks for the PR @marcelomilera! I'll take a look!

File: herokux/provider.go
Comment (marcelomilera) [reply herokux/provider.go]: Updated

Comment (davidji99) [reply herokux/provider.go]: Thank you!

**Comment by marcelomilera** [comment]
I added some tests for the new data source. Thanks for the reminder.

File: go.sum
Comment (davidji99) [inline go.sum:224]: Hmm this shouldn't be added to `go.sum`. When I pull down your PR, `go mod tidy` removes this line.

File: herokux/data_source_herokux_addons.go
Comment (davidji99) [inline herokux/data_source_herokux_addons.go:37]: See my comment I left in the `docs/data-sources/addons.md` file regarding `addon_ids` and `addon_names`

File: docs/data-sources/addons.md
Comment (davidji99) [inline docs/data-sources/addons.md:36]: It would be better use a Map/hash instead of two separate attributes here and not rely on the indexes of `addon_ids` and `addon_names` to match up . I'm thinking of this sort of data structure:

```
{
  "<ADDON_UUID>": "<ADDON_NAME>"
...
}
```

With the current implementation, it would not be easily possible to determine the addon name for an addon id.

File: api/platform/app.go
Comment (davidji99) [inline api/platform/app.go:31]: Since you aren't using any variants of the Platform API for your data source, you don't need to define a method and the associated `struct` for `AddonResponse`. You can use `heroku-go` here via the `Client` struct defined in `/api/api.go` in the resource file.

Also while this provider (and the api pkg) use my fork of [`heroku-go`](github.com/davidji99/heroku-go) in order to accomodate Platform API variants, the fork is up-to-date with [upstream](github.com/heroku/heroku-go).

File: herokux/data_source_herokux_addons.go
Comment (davidji99) [inline herokux/data_source_herokux_addons.go:47]: Continuing from my comment I left on the `api/platform/app.go` file, please make use of the `heroku-go` client here.

For example: (this is incomplete but a start on what I'm looking for)
```suggestion
	config := meta.(*Config)
	client := config.PlatformAPI
	var diags diag.Diagnostics
	
	appID := getAppID(d)
	if v, ok := d.GetOk("addon_service_name"); ok {
		addons, listErr :=  client.AddOnListByApp(context.TODO(), appID, &v5.ListRange{Descending: true})
		if listErr != nil {
			diags = append(diags, diag.Diagnostic{
				Severity: diag.Error,
				Summary:  fmt.Sprintf("Unable to retrieve addons for app %s", appID),
				Detail:   listErr.Error(),
			})
			return diags
		}
	} else {
		
	}
```

**Review by davidji99 — CHANGES_REQUESTED** [review]
Overall a good start! As always, thanks for the contribution(s)! I've left some comments below.

One thing with this proposed resource is that `herokux_addons` doesn't make use of any undocumented APIs/Platform API variants or completely alter the behavior of any existing resources in the `heroku` provider. `herokux_addons` could reside in the `heroku` provider, but I'm happy accept `herokux_addons` to this provider once the comments I've left are addressed in subsequent commits. I don't have a strong opinion here so I'll leave it up to you to decide to stay the course or move this PR to the [heroku repoisitory](https://github.com/heroku/terraform-provider-heroku). Full disclosure, these days I am the primary maintainer on the Heroku provider, but I'm not the sole maintainer.

File: go.sum
Comment (marcelomilera) [reply go.sum]: Fixed

File: docs/data-sources/addons.md
Comment (marcelomilera) [reply docs/data-sources/addons.md]: Great suggestion! using a Map now

File: api/platform/app.go
Comment (marcelomilera) [reply api/platform/app.go]: Nice catch, removed this code in favor of `heroku-go`

File: herokux/data_source_herokux_addons.go
Comment (marcelomilera) [reply herokux/data_source_herokux_addons.go]: Fixed, using `heroku-go` client now.

File: herokux/data_source_herokux_addons.go
Comment (marcelomilera) [reply herokux/data_source_herokux_addons.go]: Updated

**Comment by davidji99** [comment]
Awesome thanks for the changes!

**Review by davidji99 — APPROVED** [review]

**Pull request merged**
